libipl/p10/ipl0.C

```diff
@@ -893,70 +893,54 @@ static int ipl_set_ref_clock(void)
 	struct pdbg_target *proc;
 	int rc = 0;
 	uint8_t clock_count = 0;
-
+	fapi2::ReturnCode fapirc;
 	if (ipl_type() == IPL_TYPE_MPIPL)
 		return -1;
 
 	ipl_log(IPL_INFO, "Istep: set_ref_clock: started\n");
 
```

Comment:
We can call ipl_is_master_proc and skip if not master here in the existing code right?

Reply:
jayanth want to check if master proc is present and is functional at the beginning of the method and return error if not present. Checking for master proc present is a for loop, and for clock initialization we have another for loop. To avoid 2 for loops made this change

ipl_check_functional_master is a for loop which checks if a given proc is master or not.

we used to have
for ( proc: procList)
{
     ipl_is_master_proc(proc); // insider we have for loop
     if yes
         peform clock initalization
}

now removed the need of 2 for loops.

now I added get master functional proc method which is a for loop, once we get master proc we perform clock initialization or clock test else return error.